src/main/java/com/automation/school/ad_main/Collection/UserOrder.java

```diff
@@ -0,0 +1,80 @@
+package com.automation.school.ad_main.Collection;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserOrder  implements Order{
+
+    private long id; //(generate automatically in constructor )
+    public LocalDateTime date; //(set as current day directly in constructor)
+    public User user;
+    public List<Item> items;
+    private static int nextUniqueId = 0;
+
+
+    // public  User newUser;
+    //public  Item newItem;
+    //get the value of id
+    public UserOrder() {
+        this.id = getId();
+        this.date = LocalDateTime.now();
+        this.items = new ArrayList<>();
+        //this.user = user;
+    }
+
+    public int getId() {
```

Comment:
why id is long, but nextUniqueId is int? they have the same scope and diff.data type.
once again, if you call this method more then once - it will always change objects id, what is wrong from logical point of view.

Reply:
Logic was refactored , both var is long now
private long companyId;
 public static long companyCurrentId = 0;